src/overlays/actors/ovl_En_Wdhand/z_en_wdhand.c

```diff
@@ -105,63 +175,679 @@ static DamageTable D_80AF64F4 = {
+}
+
+void EnWdhand_Update(Actor* thisx, PlayState* play) {
+    EnWdhand* this = THIS;
+
+    EnWdhand_UpdateDamage(this, play);
+
+    this->actionFunc(this, play);
+
+    if (this->collider.base.atFlags & AT_ON) {
+        CollisionCheck_SetAT(play, &play->colChkCtx, &this->collider.base);
+    }
+    if (this->collider.base.acFlags & AC_ON) {
+        CollisionCheck_SetAC(play, &play->colChkCtx, &this->collider.base);
+    }
+    if (this->collider.base.ocFlags1 & OC1_ON) {
+        CollisionCheck_SetOC(play, &play->colChkCtx, &this->collider.base);
+    }
+}
+
+void EnWdhand_UpdateColliderLocationsForLimb(EnWdhand* this, s32 limbIndex, Vec3f* limbPos) {
+    EnWdhand_SetTransform(this, limbIndex, true, limbPos);
+    Matrix_MultVecY(575.0f, limbPos);
+    EnWdhand_Vec3fToVec3s(&this->collider.elements[2 * limbIndex + 0].dim.worldSphere.center, limbPos);
+    Matrix_MultVecY(1725.0f, limbPos);
+    EnWdhand_Vec3fToVec3s(&this->collider.elements[2 * limbIndex + 1].dim.worldSphere.center, limbPos);
+    Matrix_MultVecY(2300.0f, limbPos);
+}
+
+void EnWdhand_Draw(Actor* thisx, PlayState* play) {
+    EnWdhand* this = THIS;
+    Vec3f limbPos;
+    Gfx* dl;
+    s32 limbIndex;
+    s32 i;
+
+    OPEN_DISPS(play->state.gfxCtx);
+
+    dl = POLY_OPA_DISP;
+
+    gSPDisplayList(&dl[0], gSetupDLs[25]);
+    gSPMatrix(&dl[1], Matrix_NewMtx(play->state.gfxCtx), G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+    gSPDisplayList(&dl[2], gDexihandBaseDL);
+
+    Matrix_MultVecY(300.0f, &limbPos);
+
+    dl = &dl[3];
+
+    limbIndex = this->limbIndexBeforeCut + 1;
+    for (i = 0; i < limbIndex; i++) {
+        Matrix_Push();
+        EnWdhand_UpdateColliderLocationsForLimb(this, i, &limbPos);
+        gSPMatrix(dl++, Matrix_NewMtx(play->state.gfxCtx), G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+        gSPDisplayList(dl++, gDexihandArmSegmentDL);
+        Matrix_Pop();
+    }
+
+    Math_Vec3f_Copy(&this->endPoints[EN_WDHAND_CUT_POINT], &limbPos);
+    Math_Vec3f_Copy(&limbPos, &this->endPoints[EN_WDHAND_HAND_POINT]);
+
+    if (this->actor.world.rot.x != 0) {
+        Matrix_RotateZYX(this->actor.world.rot.x, this->actor.world.rot.z, 0, MTXMODE_NEW);
+        Matrix_RotateYS(-this->actor.world.rot.z, MTXMODE_APPLY);
+        Matrix_Mult(&this->relativeToWorldTransform, MTXMODE_APPLY);
+        Matrix_Scale(this->actor.scale.x, this->actor.scale.y, this->actor.scale.z, MTXMODE_APPLY);
+    }
+
+    for (i = this->limbIndexAfterCut; i < EN_WDHAND_NUM_SEGMENTS; i++) {
+        Matrix_Push();
+        EnWdhand_UpdateColliderLocationsForLimb(this, i, &limbPos);
+        gSPMatrix(dl++, Matrix_NewMtx(play->state.gfxCtx), G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+        gSPDisplayList(dl++, gDexihandArmSegmentDL);
+        Matrix_Pop();
+    }
+
+    EnWdhand_SetTransform(this, EN_WDHAND_NUM_SEGMENTS, true, &limbPos);
+
+    if (this->globalLimbScaleFactor > 1.0f) {
+        s32 pad;
+        f32 var_fv1 = CLAMP_MAX(this->globalLimbScaleFactor, 1.5f);
```

Comment:
Nit: seems like you could just name this variable `globalLimbScaleFactor` or `clampedGlobalLimbScaleFactor`